### PR TITLE
Add ciflow config and change jinja2 templates

### PR DESCRIPTION
**Pull request by zhouzhuojie**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #61886
* #61879

This PR is rolling out at the `1. Manual Phase`.

```
#       Rollout Strategy:
#       1. Manual Phase
#          step 1. Add 'ciflow/default' label to the PR
#          step 2. Once there's an [unassigned] event from PR, it should rerun
#          step 3. Remove 'ciflow/default' label
#          step 4. Trigger the [unassigned] event again, it should not rerun
#       2. Probot Phase 1 (manual on 1 workflow)
#          step 1. Probot automatically add labels based on the context
#          step 2. Manually let probot trigger [unassigned] event
#       4. Probot Phase 3 (auto on 1 workflows)
#          step 1. Modify the workflows so that they only listen on [unassigned] events
#          step 2. Probot automatically adds labels automatically based on the context
#          step 3. Probot automatically triggers [unassigned] event
#       4. Probot Phase 3 (auto on many workflows)
#          step 1. Enable it for all workflows
```

Differential Revision: [D29808366](https://our.internmc.facebook.com/intern/diff/D29808366)

### CODE REVIEW

File: .github/scripts/generate_ci_workflows.py [hunk 1]
```diff
@@ -26,6 +27,61 @@
...
+            for label in self.labels
+        ]
+        self.root_job = f'''
+  {self.root_job_name}:
+    runs-on: ubuntu-1804
+    if: (github.event_name != 'pull_request') || ({" && ".join(conditions)})
+    steps:
+      - name: noop
+        run: echo running {self.root_job_name}
+'''
```
Comment (samestep) [inline .github/scripts/generate_ci_workflows.py:72]: generating JavaScript code like this is kinda spooky but if it works ¯\_(ツ)_/¯

Comment (zhouzhuojie) [reply .github/scripts/generate_ci_workflows.py:72]: yup, actually, will move to the template, as it's better sitting there

File: .github/scripts/generate_ci_workflows.py [hunk 2]
```diff
@@ -71,7 +131,7 @@ def generate_workflow_file(self, workflow_template: jinja2.Template) -> None:
         with open(output_file_path, "w") as output_file:
             GENERATED = "generated"
             output_file.writelines([f"# @{GENERATED} DO NOT EDIT MANUALLY\n"])
-            output_file.write(workflow_template.render(**asdict(workflow)))
+            output_file.write(workflow_template.render(asdict(workflow)))
```
Comment (samestep) [inline .github/scripts/generate_ci_workflows.py:134]: I don't quite understand this change; could you explain?

File: .github/workflows/periodic-pytorch-libtorch-linux-xenial-cuda11.3-cudnn8-py3.6-gcc7.yml [hunk 3]
```diff
@@ -98,7 +99,7 @@ jobs:
 
   build:
     runs-on: linux.2xlarge
-    needs: calculate-docker-image
+    needs: [calculate-docker-image, ]
```
Comment (samestep) [inline .github/workflows/periodic-pytorch-libtorch-linux-xenial-cuda11.3-cudnn8-py3.6-gcc7.yml:102]: is this syntax ok? I'm not a yaml expert

Comment (zhouzhuojie) [reply .github/workflows/periodic-pytorch-libtorch-linux-xenial-cuda11.3-cudnn8-py3.6-gcc7.yml:102]: yes, trailing comma is allowed in yaml

File: .github/scripts/generate_ci_workflows.py (hunk 2 above)
Comment (zhouzhuojie) [reply .github/scripts/generate_ci_workflows.py:128]: it also takes a dict object, so saves some time and no need to convert to `**kwargs`

File: .github/scripts/generate_ci_workflows.py (hunk 1 above)
Comment (zhouzhuojie) [reply .github/scripts/generate_ci_workflows.py]: updated, good point of keeping python clean

**Comment by zhouzhuojie** [comment]
@zhouzhuojie has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29808366).

**Comment by zhouzhuojie** [comment]
@zhouzhuojie has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29808366).

**Review by seemethere — APPROVED** [review]
LGTM

**Comment by zhouzhuojie** [comment]
@zhouzhuojie has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29808366).

**Pull request closed without merging**